routes/approvals.js

```diff
@@ -40,11 +40,36 @@ approvals.getApprovals = function(app, res, loggedInUserId) {
     var userInfo = users.loadUser(app, loggedInUserId);
     if (!userInfo)
         return res.status(403).jsonp({ message: 'Not allowed' });
-    if (!userInfo.admin)
+    if (!userInfo.admin && !userInfo.approver)
         return res.status(403).jsonp({ message: 'Not allowed' });
-    
+
     var approvalInfos = approvals.loadApprovals(app);
+    approvalInfos = approvalInfos.filter(function(approval) {
+      if(userInfo.admin) return true; //Show all approvals for admin
+      if(!userInfo.groups) return false; //Approver is not attached to a group (un-likely)
+      if(!approval.api.requiredGroup) return false; //Api is not attached any group
+
+      var groups = utils.loadGroups(app);
+      groups = groups.groups;
+      if(!groups) return false;
+      var usrGroupsHash = {};
+      for(var i=0; i< userInfo.groups.length; i++){
+        usrGroupsHash[userInfo.groups[i]] = userInfo.groups[i];
+      }
+      for(var i=0; i< groups.length; i++){
+        if(groups[i].id==usrGroupsHash[groups[i].id]){
+          var alt_ids = groups[i].alt_ids;
+          if(alt_ids){
+            for(var j=0; j< alt_ids.length; j++){
+              usrGroupsHash[alt_ids[j]] = alt_ids[j];
+            }
+          }
+        }
+      }
+      //if group id or alt_id of approvar's group matches with requiredGroup of an API, return happy
+      return (usrGroupsHash[approval.api.requiredGroup]==approval.api.requiredGroup);
```

Comment:
This entire predicate function is a little too complicated for what it does, in my opinion. It's very confusing you build up a map which looks like this:

```
usrGroupsHash = {
  admin: "admin",
  somegroup: "somegroup",
  dev: "dev"
}
```

It's not a "hash", a "hash" is a calculated value based on some data, like an md5 or sha1 hash, but this is just a map/dictionary, so that's a little confusing. The second thing which is confusing is that you're checking that the string is (a) present and (b) matches the `requiredGroup`. Is this necessary, or wouldn't it be enough to check whether the `requiredGroup` is present in the `usrGroupsHash`?

Please rename `usrGroupsHash` to `userGroupMap` or similar, and try to stick to the coding conventions of the surrounding code, e.g.

* `if` statements on two lines
* Spaces around operators/assignments

Sorry for nitpicking. Thanks for your contributions!